### PR TITLE
respect resource key + default serializer

### PR DESCRIPTION
**Pull request by yaapis**

### CODE REVIEW

File: src/Skeleton/BaseController.php [hunk 1]
```diff
@@ -274,7 +299,7 @@ protected function setStatusCode($statusCode)
      */
     protected function respondWithItem($item)
     {
-        $resource = new Item($item, $this->transformer);
+        $resource = new Item($item, $this->transformer, $this->resourceKey);
```
Comment (arrilot) [inline src/Skeleton/BaseController.php:304]: This looks wrong to me.
Shouldn't it be `str_singular($this->resourceKey)`?

Comment (arrilot) [inline src/Skeleton/BaseController.php:304]: I've also would have gone another way - `$resourceKey='car'` and use `str_plural`

File: src/Skeleton/BaseController.php [hunk 2]
```diff
@@ -102,6 +111,16 @@ public function __construct(Request $request)
     abstract protected function transformer();
 
     /**
+     * Serializer for the current model.
+     *
+     * @return \League\Fractal\Serializer\SerializerAbstract
+     */
+    protected function serializer()
+    {
+        return new Serializer();
```
Comment (arrilot) [inline src/Skeleton/BaseController.php:120]: The default serializer should be DataArraySerializer for BC

File: src/Skeleton/BaseController.php (hunk 1 above)
Comment (yaapis) [inline src/Skeleton/BaseController.php:304]: There are some cases when you will have different keys on index, update, create methods.
Thats why its better to leave respondWithItem and respondWithCollection just with $this->resourceKey, and singular | plural - in create + update methods.

In my case - I can easy rewrite create method to make my key singular.

And, yeah, I think y're right - resourceKey should be singular by default.

**Comment by arrilot** [comment]
Thanks.
I've tweaked a lot of things thought :)
You can steal override anything in your own app if you need it,

**Pull request merged**